docs/en/query_language/syntax.md

```diff
@@ -26,23 +26,33 @@ SQL-style and C-style comments are supported.
 SQL-style comments: from `--` to the end of the line. The space after `--` can be omitted.
 Comments in C-style: from `/*` to `*/`. These comments can be multiline. Spaces are not required here, either.
 
-## Keywords
+## Keywords {#syntax-keywords}
 
-Keywords (such as `SELECT`) are not case-sensitive. Everything else (column names, functions, and so on), in contrast to standard SQL, is case-sensitive. Keywords are not reserved (they are just parsed as keywords in the corresponding context).
+Keywords (such as `SELECT`) are not case-sensitive. Everything else (column names, functions, and so on), in contrast to standard SQL, is case-sensitive.
 
-## Identifiers
+Keywords are not reserved (they are just parsed as keywords in the corresponding context). If you use [identifiers](#syntax-identifiers) the same as the keywords, enclose them into quotes, for example, `"column_name"`. The query `SELECT "FROM" FROM table_name` is valid if the table `table_name` has column with the name `"FROM"`.
 
-Identifiers (column names, functions, and data types) can be quoted or non-quoted.
-Non-quoted identifiers start with a Latin letter or underscore, and continue with a Latin letter, underscore, or number. In other words, they must match the regex `^[a-zA-Z_][0-9a-zA-Z_]*$`. Examples: `x, _1, X_y__Z123_.`
+## Identifiers {#syntax-identifiers}
 
-Quoted identifiers are placed in reversed quotation marks `` `id` `` (the same as in MySQL), and can indicate any set of bytes (non-empty). In addition, symbols (for example, the reverse quotation mark) inside this type of identifier can be backslash-escaped. Escaping rules are the same as for string literals (see below).
-We recommend using identifiers that do not need to be quoted.
+Identifiers are:
+
+- Cluster, database, table, partition and column names;
+- Functions;
+- Data types;
+- [Expression aliases](#syntax-expression_aliases).
+
+Identifiers can be quoted or non-quoted. It is recommended to use non-quoted identifiers.
+
+Non-quoted identifiers must match the regex `^[a-zA-Z_][0-9a-zA-Z_]*$` and can not be equal to [keywords](#syntax-keywords).  Examples: `x, _1, X_y__Z123_.`
+
+If you want to use identifiers the same as keywords or you want to use any symbols in identifiers, quote it. Use `"id"` or `` `id` `` syntax.
```

Comment:
* any symbols -> other symbols
* quote it. Use `"id"` or `` `id` `` syntax. -> quote it using double quotes or backticks, for example `"id"` or `` `id` ``.